rewriter/src/main/java/com/facebook/coresql/rewriter/approxpercentile/ApproxPercentilePatternMatcher.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.rewriter.approxpercentile;
+
+import com.facebook.coresql.parser.AstNode;
+import com.facebook.coresql.parser.Select;
+import com.facebook.coresql.parser.SqlParserDefaultVisitor;
+import com.facebook.coresql.rewriter.PatternMatcher;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.facebook.coresql.parser.SqlParserTreeConstants.JJTARGUMENTLIST;
+import static com.facebook.coresql.parser.SqlParserTreeConstants.JJTFUNCTIONCALL;
+import static com.facebook.coresql.parser.SqlParserTreeConstants.JJTIDENTIFIER;
+import static com.facebook.coresql.parser.SqlParserTreeConstants.JJTSELECTLIST;
+import static com.facebook.coresql.parser.SqlParserTreeConstants.JJTUNSIGNEDNUMERICLITERAL;
+import static java.util.Objects.requireNonNull;
+
+public class ApproxPercentilePatternMatcher
+        extends SqlParserDefaultVisitor
+        implements PatternMatcher
+{
+    private final Map<AstNode, Map<String, List<AstNode>>> selectMap; // A map of SelectNode to a map of each unique first arg that has duplicate APPROX_PERCENTILE nodes
+    private boolean nonLiteralSecondArgFound;
+
+    public ApproxPercentilePatternMatcher()
+    {
+        selectMap = new HashMap<>();
+    }
+
+    private boolean isLiteral(AstNode node)
+    {
+        return node.getId() == JJTUNSIGNEDNUMERICLITERAL;
+    }
+
+    private void processSelectItem(AstNode node, Map<String, List<AstNode>> map)
+    {
+        AstNode functionCall = node.GetFirstChildOfKind(JJTFUNCTIONCALL);
+        if (functionCall == null) {
+            return;
+        }
+        AstNode identifier = functionCall.GetFirstChildOfKind(JJTIDENTIFIER);
+        if (identifier == null || !identifier.GetImage().equalsIgnoreCase("APPROX_PERCENTILE")) {
```

Comment:
Add a utilitiy method - isApproxPercentile and do this check there.

Reply:
Done